virt_lightning/__init__.py

```diff
@@ -37,31 +38,37 @@ def run_cmd(cmd, cwd=None):
 
 
 class LibvirtHypervisor:
-    def __init__(self, configuration):
-        conn = libvirt.open(configuration.libvirt_uri)
+    def __init__(self, libvirt_uri):
+        conn = libvirt.open(libvirt_uri)
 
         if conn is None:
             error_tpl = "Failed to open connection to {uri}"
-            print(error_tpl.format(uri=configuration.libvirt_uri))
+            print(error_tpl.format(uri=libvirt_uri))
             exit(1)
 
         self.conn = conn
-        self.configuration = configuration
-        self.init_network()
         self._last_free_ipv4 = None
-        self.pool = self.conn.storagePoolLookupByName(self.configuration.storage_pool)
-        self.get_storage_dir()
         self.wait_for = []
         self.kvm_binary = self.find_kvm_binary()
 
+    def get_arch(self):
+        caps = self.conn.getCapabilities()
+        root = ET.fromstring(caps)
+        return root.find("./host/cpu/arch").text
+
+    def get_type(self):
```

Comment:
This is like `@propety` too.

Reply:
I didn't know about this decorator. I rename the method just `type()`.